tests/test_errors.py

```diff
@@ -455,6 +475,38 @@ def finalize(self, cfg: Any) -> None:
         ),
         id="dict[int,Any]:mistyped_key",
     ),
+    pytest.param(
+        Expected(
+            create=lambda: DictConfig({}, key_type=float),
+            op=lambda cfg: cfg.get("foo"),
+            exception_type=KeyValidationError,
+            msg=dedent(
+                """\
+                Key foo (str) is incompatible with (float)
+                    full_key: foo
+                    object_type=dict"""
+            ),
+            key="foo",
+            full_key="foo",
+        ),
+        id="dict[float,Any]:mistyped_key",
+    ),
+    pytest.param(
+        Expected(
+            create=lambda: DictConfig({}, key_type=bool),
+            op=lambda cfg: cfg.get("foo"),
+            exception_type=KeyValidationError,
+            msg=dedent(
+                """\
+                Key foo (str) is incompatible with (bool)
+                    full_key: foo
+                    object_type=dict"""
+            ),
```

Comment:
here too.

Reply:
Done in https://github.com/omry/omegaconf/pull/484/commits/d9efe4478dbef91c69d59f2721b4e758babd3238